src/app/routes/secrets/SecretsList.js

```diff
@@ -683,90 +652,55 @@ const _mapStateToProps = (state, ownProps) => {
     }
     const type = ((location || {}).state || {}).type || null;
     const isDynamicSecret = Constants.DYNAMIC_ENGINES.some(engine => engine === type);
-    let dynamicSecretRole = [];
-    let secretsList = [];
-    let requestId = null;
-    // TODO CONSOLIDATE DYNAMIC AND STANDARD REQUEST CODE?
-    // ☝️ 100%. Yes. This hurts my eyes. -JH
-    if (isDynamicSecret) {
-        dynamicSecretRole = (secretsPaths.secrets || []).map(role => {
-            const {capabilities, name} = role;
-            const engineNameRole = mountData && `${mountData.name.slice(0, -1)}/${name}`;
-            let isApproved = false;
-            let isOpened = false;
-            const requiresRequest = !capabilities.includes('read');
-            let secondaryText = 'Request type: Dynamic Request';
-            const activeDynamicRequest = secretsRequests.find(request => request.path === engineNameRole);
-            if (activeDynamicRequest) {
-                const {approved, authorizations, creationTime, opened, requestId: reqId} = activeDynamicRequest;
-                isApproved = approved && !opened;
-                requestId = reqId;
+    const secretsList = (secretsPaths.secrets || []).map((secret) => {
+        const {capabilities, data = {}, name} = secret;
+        const {wrap_info: wrapInfo} = data;
+        const currentPath = path ? `${path}/${name}` : name;
+        const mountPath = `${mount}/${currentPath}`;
+        const isWrapped = !!wrapInfo;
+        let isApproved = false;
+        let isOpened = false;
+        let isDSCanOpen = false;
+        const requiresRequest = !capabilities.includes('read') && !name.endsWith('/') || isWrapped;
+        let activeRequest;
+        let secondaryText;
+        let authorizationsText;
+        if (requiresRequest) {
+            secondaryText = `Request type: ${isWrapped ? 'Control Groups' : isDynamicSecret ? 'Dynamic Request' : 'Standard Request'}`;
+            // Check for any active requests.
+            const secretsPath = `${mount}${isV2 ? '/data/' : '/'}${currentPath}`;
+            activeRequest = secretsRequests.find((request) => request.path === secretsPath);
+            if (activeRequest) {
+                const {approved, authorizations, creationTime, opened} = activeRequest;
                 isOpened = opened;
+                isDSCanOpen = isDynamicSecret ? !isOpened : true;
+                isApproved = approved && isDSCanOpen;
```

Comment:
Is this correct? What about approved Control Group request?

Reply:
if its control groups / standard request that will render it to true. In this part `isDynamicAndCanOpen = isDynamicSecret ? !isOpened : true;`